integration/mountkexec_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package integration
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/cp"
+	"github.com/u-root/u-root/pkg/golang"
+	"github.com/u-root/u-root/pkg/uroot"
+	"github.com/u-root/u-root/pkg/uroot/builder"
+	"github.com/u-root/u-root/pkg/uroot/initramfs"
+)
+
+// genISO generates an iso containing a kernel and initramfs to be used for testing mount and kexec
+func genISO() (err error) {
+	// Copy kernel to tmpDir for tests involving kexec.
+	tmpDir, err := ioutil.TempDir("", "uroot-integration")
+	if err != nil {
+		return err
+	}
+	kernel := filepath.Join(tmpDir, "kernel")
+	if err := cp.Copy(os.Getenv("UROOT_KERNEL"), kernel); err != nil {
+		return err
+	}
+
+	// OutputFile
+	newOutputFile := filepath.Join(tmpDir, "initiso.cpio")
+	w, err := initramfs.CPIO.OpenWriter(log.New(os.Stderr, "", 0), newOutputFile, "", "")
+	if err != nil {
+		return err
+	}
+
+	cmds := []string{"github.com/u-root/u-root/cmds/*"}
+	env := golang.Default()
+	env.CgoEnabled = false
+	env.GOARCH = TestArch()
+
+	// Build u-root
+	opts := uroot.Opts{
+		Env: env,
+		Commands: []uroot.Commands{
+			{
+				Builder:  builder.BusyBox,
+				Packages: cmds,
+			},
+		},
+		TempDir:      tmpDir,
+		BaseArchive:  uroot.DefaultRamfs.Reader(),
+		OutputFile:   w,
+		DefaultShell: "elvish",
+	}
+	if err := uroot.CreateInitramfs(log.New(os.Stderr, "", 0), opts); err != nil {
+		return err
+	}
+
+	if err = exec.Command("genisoimage", "-o", "/tmp/tempdata.iso", tmpDir).Run(); err != nil {
```

Comment:
use CombinedOutput, not Run, so you can have some control over where output goes. This is a test and it is not allowed to generate lots of output in an uncontrolled way.

Reply:
Done